DATASET_VERSIONING.md

```diff
@@ -0,0 +1,76 @@
+# Dataset versioning (DRAFT)
+
+DSO API provides option to serve multiple versions of same dataset via API and Database.
+
+This document aims to cover technical details on how DSO API behaves in different scenarios, 
+depending on Dataset model configuration.
+
+
+## How dataset version is defined
+
+`Dataset` model exposes 2 parameters to control versioning:
+
+ - `Dataset.version`: string, formatted according to [Semantic Versioning standard](https://semver.org/) e.g. `0.1.0` 
+   (`<major> "." <minor> "." <patch>`).
+ - `Dataset.is_default_version`: boolean, defaults to `True`
+
+Combination of these 2 parameters defines how API and Database behave.
+
+
+## Datasets without version definition and default versions of datasets
+
+All datasets without version definition in `Dataset.version` or with `Dataset.is_default_version == True` 
+will be treated as not versioned and all tables within dataset will have API endpoint set to 
+
+> `/v1/<dataset.url_prefix>/<dataset.id>/<table.name>/`
+
+Database tables of this dataset will be prefixed with `<dataset.id>_`
+
+
+## Datasets with non-default versions
+
+Any dataset wit `version` defined and `is_default_version` is set to `False` will have API endpoint defined as:
+
+ > `/v1/<dataset.url_prefix>/<dataset.id>@<dataset.version>/<table.name>/`
```

Comment:
Does the API endpoint still start with `/v1`? If so, what does the `v1` stand for in relation the `<dataset.version>`?

Reply:
Yes, as it is API version, not dataset version. `/v1` is expected to be changed when we change JSON structure drastically.

Reply:
Nevermind. It's already clear to me.